internal/shim-sgx/src/handler/mod.rs

```diff
@@ -164,26 +207,38 @@ impl guest::Handler for Handler<'_> {
         fd: c_int,
         offset: off_t,
     ) -> sallyport::Result<NonNull<c_void>> {
-        Ok(NonNull::new(crate::heap::HEAP.write().mmap::<c_void>(
-            addr.map(|v| v.as_ptr() as usize).unwrap_or(0),
-            length,
-            prot,
-            flags,
-            fd,
-            offset,
-        )?)
-        .unwrap())
+        let addr = addr.map(|v| v.as_ptr() as usize).unwrap_or(0);
+
+        if addr != 0 || length == 0 || fd != -1 || offset != 0 {
+            return Err(EINVAL);
```

Comment:
@rvolosatovs would say, that we should return `ENOTSUP` :smile:

Reply:
fixed